src/c_bindings/c_wrapper.cpp

```diff
@@ -7,40 +7,57 @@
 extern "C" {
     // C wrapper function
     const char* create_discriminant_wrapper(const uint8_t* seed, size_t seed_size, int length) {
-        std::vector<uint8_t> seedVector(seed, seed + seed_size);
-        integer result = CreateDiscriminant(seedVector, length);
+        try {
+            std::vector<uint8_t> seedVector(seed, seed + seed_size);
+            integer result = CreateDiscriminant(seedVector, length);
 
-        // Serialize the 'result' to a string
-        std::string resultStr = result.to_string();
+            // Serialize the 'result' to a string
+            std::string resultStr = result.to_string();
```

Comment:
this looks a bit fishy. `integer` is a GMP big int. converting it to a decimal ascii representation to return seems like an expensive detour. Also, do you know if there's an upper limit on the size of these integers? If so, you could pass in a buffer to be filled-in instead of returning a new heap allocation. But I suspect there is not upper limit.